WIP - Retarget to 2021.2

maartenba opened the pull request:
Do not merge yet, work in progress.

* Need to pin to proper EAP packages in `idea` 
* Can't compile - https://github.com/JetBrains/gradle-intellij-plugin/issues/674

Fixes #493
Fixes #494

maartenba committed:
Retarget to 2021.2 EAP

- Bump Rider and IntelliJ plugins SDK version
- Bump IntelliJ Gradle plugin version
- Bump Gradle version
- Fix compilation after bump Rider and IntelliJ SDK to 212 version
- Rebuild Gradle wrapper files
- Retarget ReSharper projects to Net 472 version to provide SDK compartibility
- Update build scripts

sdubov commented on BuildPlugin:
```diff
@@ -34,11 +34,11 @@ tc_open "Initializing build script"
     IJ_VERSION_LATEST=$IJ_VERSION_MAJOR.$IJ_VERSION_MINOR \
                       && [[ -z $IJ_VERSION_MINOR ]] && IJ_VERSION_LATEST=$IJ_VERSION_MAJOR
-    IJ_SCALA_VERSION_LATEST=2021.1.15
+    IJ_SCALA_VERSION_LATEST=2021.2.15
```
I found the latest Scala plugin version is `2021.2.9`.

sdubov reviewed (approved):
Need to fix BuildPlugin script that cause failures in Plugin build and I think we can bump all related versions again )

maartenba commented:
Still running into JetBrains/gradle-intellij-plugin#674 :-(

sdubov committed:
Fix Gralde test tasks that miss prepareTestingSandbox task dependency call

- Fix task to run Rider tests
- Update build scrips and version of local dependencies

maartenba commented:
Thanks @sdubov!
